src/common/slow-query-recorder/src/slow_query_recorder.rs

```diff
@@ -446,13 +472,34 @@ impl SlowQueryEventHandler {
             },
         ]
     }
+
+    fn is_sql_query_statement(&self, query: &str, query_ctx: QueryContextRef) -> bool {
+        match ParserContext::create_with_dialect(
```

Comment:
Parsing SQL queries to check their type is too expensive. A simpler, though less robust, approach is to check if the query string starts with `select` or `with`.